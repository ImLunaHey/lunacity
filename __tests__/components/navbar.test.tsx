import { render } from '@testing-library/react';
import { SessionProvider } from 'next-auth/react';

import Navbar from '@app/components/navbar';

import '@testing-library/jest-dom';

import type { Session } from 'next-auth';
import React from 'react';
import { replaceDynamicIds } from '../__utils__/replace-dynamic-ids';

// Mock next-auth with defaults
jest.mock('next/config', () => ({
  __esModule: true,
  default: () => ({
    publicRuntimeConfig: {
      APP_URL: 'http://localhost:3000',
      WS_URL: 'ws://localhost:3001',
    },
  }),
}));

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({
    asPath: '/',
  })),
}));

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn(() => ({
    t: jest.fn((text: string) => text),
    ready: true,
  })),
}));

// Mock notifications component
jest.mock('../../src/components/notifications', () => ({
  __esModule: true,
  Notifications: () => <div>__NOTIFICATIONS__MOCK__</div>,
}));

describe('Navbar', () => {
  it('renders nothing when unauthenticated', () => {
    const { container } = render(
      <SessionProvider session={null}>
        <Navbar />
      </SessionProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the navbar when authenticated', () => {
    const mockSession = {
      expires: '1',
      user: {
        id: '123',
        page: {
          id: '456',
          displayName: '@test-user',
          handle: 'test-user',
          image: '',
          deactivatedTimestamp: null,
          description: '',
          followerCount: 0,
          official: false,
          ownerId: '123',
          userId: '123',
        },
      },
    } satisfies Session;

    const { container } = render(
      <SessionProvider session={mockSession}>
        <Navbar />
      </SessionProvider>
    );
    expect(replaceDynamicIds(container)).toMatchSnapshot();
  });
});
