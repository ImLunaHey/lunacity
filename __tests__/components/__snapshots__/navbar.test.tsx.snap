// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar it renders nothing when unauthenticated 1`] = `<div />`;

exports[`Navbar it renders the navbar when authenticated 1`] = `
<div>
  <nav
    class="nextui-c-hhqfap nextui-c-hhqfap-ehawcb-variant-sticky nextui-c-hhqfap-beFEYN-maxWidth-lg nextui-c-hhqfap-hemDFs-isBordered-true nextui-c-hhqfap-kAwfsS-disableShadow-false nextui-c-hhqfap-gqZKcP-disableBlur-false nextui-c-hhqfap-iPJLV-css nextui-navbar"
  >
    <div
      class="nextui-c-egxOLa nextui-navbar-container"
    >
      <span
        class="nextui-c-fyNvBR nextui-c-PJLV nextui-c-fyNvBR-illEebI-css nextui-navbar-brand"
      >
        <a
          href="/"
        >
          <svg
            class="h-6 w-6"
            version="1.1"
            viewBox="806.87 667.45 530.58 686.57"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
             
            <g
              id="document"
              transform="matrix(1,0,0,1,1024.0,1024.0)"
            >
               
              <path
                d="M313.443,165.007 C313.443,74.2798 240.225,0 148.436,0 L-43.0996,0 C-68.0364,0 -85.0146,6.89741 -98.8095,19.6311 C-111.013,32.3648 -201.74,131.581 -201.74,131.581 C-211.29,141.132 -217.127,152.274 -217.127,165.007 C-217.127,177.741 -211.29,188.883 -201.74,198.433 C-201.74,198.433 -112.074,296.589 -98.2789,310.383 C-85.5452,323.117 -67.5058,330.015 -42.569,330.015 L148.436,330.015 C239.164,330.015 313.443,256.265 313.443,165.007 Z "
                fill="#ffd45c"
                fill-opacity="1.00"
              />
               
              <path
                d="M217.127,-191.536 C217.127,-282.263 143.908,-356.543 52.1193,-356.543 L-43.0996,-356.543 C-68.0364,-356.543 -85.0146,-349.646 -98.8095,-336.912 C-111.013,-324.178 -201.74,-224.962 -201.74,-224.962 C-211.29,-215.411 -217.127,-204.269 -217.127,-191.536 C-217.127,-178.802 -211.29,-167.66 -201.74,-158.11 C-201.74,-158.11 -112.074,-59.9544 -98.2789,-46.1596 C-85.5452,-33.4259 -67.5058,-26.5285 -42.569,-26.5285 L52.1193,-26.5285 C142.847,-26.5285 217.127,-100.278 217.127,-191.536 Z "
                fill="#ffd45c"
                fill-opacity="1.00"
              />
               
            </g>
             
          </svg>
        </a>
        <a
          href="/"
        >
          <b
            class="nextui-c-PJLV nextui-c-PJLV-ghzjVU-hideIn-xs nextui-c-PJLV-ilaJdKq-css"
          >
            Lunacity
          </b>
        </a>
        <ul
          class="nextui-c-jklhtj nextui-c-dGYPDG nextui-c-PJLV nextui-c-PJLV-ghzjVU-hideIn-xs nextui-c-jklhtj-illCOkV-css nextui-navbar-content"
          style="--nextui--transitionDuration: 0ms; --nextui--opacity: 0; --nextui--width: 0px; --nextui--left: 0px;"
        >
          <li
            class="nextui-c-kOXsPe nextui-c-yrlUe nextui-c-PJLV nextui-c-yrlUe-kJGdUE-activeColor-default nextui-c-yrlUe-cdrGrt-isActive-true nextui-c-yrlUe-hYJzFf-underlineHeight-normal nextui-c-yrlUe-hwFizd-cv nextui-c-kOXsPe-ifcTEZe-css nextui-navbar-item"
            id="static-id-0"
          >
            <a
              class="nextui-c-kzikkm nextui-c-gulvcB nextui-c-kzikkm-hfnKRU-color-default nextui-c-kzikkm-isWIbj-animated-true nextui-c-kzikkm-iejNGLI-css nextui-link nextui-navbar-link"
              href="/"
              id="static-id-0"
              role="link"
              tabindex="0"
            >
              üè° 
            </a>
          </li>
          <li
            class="nextui-c-kOXsPe nextui-c-yrlUe nextui-c-PJLV nextui-c-yrlUe-kJGdUE-activeColor-default nextui-c-yrlUe-hYJzFf-underlineHeight-normal nextui-c-kOXsPe-ifcTEZe-css nextui-navbar-item"
            id="static-id-2"
          >
            <a
              class="nextui-c-kzikkm nextui-c-gulvcB nextui-c-kzikkm-hfnKRU-color-default nextui-c-kzikkm-isWIbj-animated-true nextui-c-kzikkm-iejNGLI-css nextui-link nextui-navbar-link"
              href="/explore"
              id="static-id-2"
              role="link"
              tabindex="0"
            >
              üåè 
            </a>
          </li>
          <li
            class="nextui-c-kOXsPe nextui-c-yrlUe nextui-c-PJLV nextui-c-yrlUe-kJGdUE-activeColor-default nextui-c-yrlUe-hYJzFf-underlineHeight-normal nextui-c-kOXsPe-ifcTEZe-css nextui-navbar-item"
            id="static-id-4"
          >
            <a
              class="nextui-c-kzikkm nextui-c-gulvcB nextui-c-kzikkm-hfnKRU-color-default nextui-c-kzikkm-isWIbj-animated-true nextui-c-kzikkm-iejNGLI-css nextui-link nextui-navbar-link"
              href="/messages"
              id="static-id-4"
              role="link"
              tabindex="0"
            >
              üì© 
            </a>
          </li>
        </ul>
      </span>
      <ul
        class="nextui-c-jklhtj nextui-c-dGYPDG nextui-c-PJLV nextui-c-jklhtj-ijztoav-css nextui-navbar-content w-full"
        style="--nextui--transitionDuration: 0ms; --nextui--opacity: 0; --nextui--width: 0px; --nextui--left: calc(var(--nextui--navbarContentItemHorizontalPadding) * 0.5 * -1);"
      >
        <li
          class="nextui-c-kOXsPe nextui-c-yrlUe nextui-c-PJLV nextui-c-yrlUe-kJGdUE-activeColor-default nextui-c-yrlUe-hYJzFf-underlineHeight-normal nextui-c-kOXsPe-iPJLV-css nextui-navbar-item w-full"
          id="static-id-6"
        >
          <form
            id="static-id-6"
          >
            <div
              class="nextui-c-iwmZdB nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-gOvjVW-size-md nextui-c-iwmZdB-ihzSmcs-css nextui-input-main-container nextui-input-main-container--normal"
              data-state="normal"
            >
              <div
                class="nextui-c-boZsAS nextui-c-boZsAS-fjttfO-isTextarea-false nextui-c-boZsAS-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input w-full"
              >
                <label
                  class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
                >
                  <span
                    class="nextui-c-PJLV nextui-c-iFHsrU nextui-input-placeholder"
                  >
                    Search...
                  </span>
                  <span
                    class="nextui-c-PJLV nextui-input-content nextui-input-content--left"
                  >
                    <svg
                      fill="none"
                      height="16"
                      viewBox="0 0 24 24"
                      width="16"
                    >
                      <path
                        d="M11.5 21a9.5 9.5 0 1 0 0-19 9.5 9.5 0 0 0 0 19ZM22 22l-2-2"
                        stroke="var(--nextui-colors-accents6)"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                  <input
                    aria-label="Search box"
                    aria-multiline="false"
                    aria-placeholder="Search..."
                    aria-readonly="false"
                    aria-required="false"
                    autocomplete="off"
                    class="nextui-c-jeuecp nextui-c-jeuecp-bVyrVm-hasLeftContent-true nextui-input nextui-input-left-content"
                    data-state="normal"
                    id="static-id-8"
                    name="query"
                    placeholder="Search..."
                    type="text"
                    value=""
                  />
                  <button
                    aria-label="clear"
                    class="nextui-c-ctQnhE nextui-input-clear-button"
                    lang="en"
                    type="button"
                  >
                    <span
                      class="nextui-c-gDSmsE nextui-c-gulvcB"
                    >
                      <svg
                        class=""
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </button>
                </label>
              </div>
              <div
                class="nextui-c-gJvMas nextui-input-helper-text-container"
              />
            </div>
          </form>
        </li>
      </ul>
      <ul
        class="nextui-c-jklhtj nextui-c-dGYPDG nextui-c-PJLV nextui-c-jklhtj-idJpuyv-css nextui-navbar-content"
        style="--nextui--transitionDuration: 0ms; --nextui--opacity: 0; --nextui--width: 0px; --nextui--left: calc(var(--nextui--navbarContentItemHorizontalPadding) * 0.5 * -1);"
      >
        <div>
          __NOTIFICATIONS__MOCK__
        </div>
        <li
          class="nextui-c-kOXsPe nextui-c-yrlUe nextui-c-PJLV nextui-c-yrlUe-kJGdUE-activeColor-default nextui-c-yrlUe-hYJzFf-underlineHeight-normal nextui-c-kOXsPe-iPJLV-css nextui-navbar-item"
          id="static-id-9"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="nextui-c-gEEnVX nextui-c-gulvcB nextui-c-gEEnVX-eeabkr-color-primary nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-gEEnVX-ifMkvH-css"
            data-state="loading"
            id="static-id-9"
            type="button"
          >
            <span
              class="nextui-avatar-bg"
            />
            <img
              class="nextui-avatar-img nextui-avatar--loading"
              data-state="loading"
              src="https://i.pravatar.cc/150?u=a042581f4e29026704d"
            />
          </button>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
